ako-gateway-api/k8s/ako_init.go

```diff
@@ -0,0 +1,450 @@
+		if err == nil {
+			resVer := meta.GetResourceVersion()
+			objects.SharedResourceVerInstanceLister().Save(key, resVer)
+		}
+		if IsValidGateway(key, gatewayObj) {
+			akogatewayapinodes.DequeueIngestion(key, true)
+		}
+	}
+
+	// HTTPRoute Section
+	httpRouteObjs, err := akogatewayapilib.AKOControlConfig().GatewayApiInformers().HTTPRouteInformer.Lister().HTTPRoutes(metav1.NamespaceAll).List(labels.Set(nil).AsSelector())
+	if err != nil {
+		utils.AviLog.Errorf("Unable to retrieve the httproutes during full sync: %s", err)
+		return err
+	}
+
+	for _, httpRouteObj := range httpRouteObjs {
+		key := lib.HTTPRoute + "/" + utils.ObjKey(httpRouteObj)
+		meta, err := meta.Accessor(httpRouteObj)
+		if err == nil {
+			resVer := meta.GetResourceVersion()
+			objects.SharedResourceVerInstanceLister().Save(key, resVer)
+		}
+		if IsHTTPRouteValid(key, httpRouteObj) {
+			akogatewayapinodes.DequeueIngestion(key, true)
+		}
+	}
+
+	// Service Section
+	svcObjs, err := utils.GetInformers().ServiceInformer.Lister().Services(metav1.NamespaceAll).List(labels.Set(nil).AsSelector())
+	if err != nil {
+		utils.AviLog.Errorf("Unable to retrieve the services during full sync: %s", err)
+		return err
+	}
+
+	for _, svcObj := range svcObjs {
+		key := utils.Service + "/" + utils.ObjKey(svcObj)
+		meta, err := meta.Accessor(svcObj)
+		if err == nil {
+			resVer := meta.GetResourceVersion()
+			objects.SharedResourceVerInstanceLister().Save(key, resVer)
+		}
+		// Not pushing the service to the next layer as it is
+		// not required since we don't create a model out of service
+	}
+
+	if sync {
+		c.publishAllParentVSKeysToRestLayer()
+	}
+	return nil
+}
+
+func (c *GatewayController) publishAllParentVSKeysToRestLayer() {
+	cache := avicache.SharedAviObjCache()
+	vsKeys := cache.VsCacheMeta.AviCacheGetAllParentVSKeys()
+	utils.AviLog.Debugf("Got the VS keys: %s", vsKeys)
+	allModelsMap := objects.SharedAviGraphLister().GetAll()
+	allModels := make(map[string]struct{})
+	vrfModelName := lib.GetModelName(lib.GetTenant(), lib.GetVrf())
+	for modelName := range allModelsMap.(map[string]interface{}) {
+		// ignore vrf model, as it has been published already
+		if modelName != vrfModelName && !lib.IsIstioKey(modelName) {
+			allModels[modelName] = struct{}{}
+		}
+	}
+	sharedQueue := utils.SharedWorkQueue().GetQueueByName(utils.GraphLayer)
+
+	for _, vsCacheKey := range vsKeys {
+		modelName := vsCacheKey.Namespace + "/" + vsCacheKey.Name
+		delete(allModels, modelName)
+		utils.AviLog.Infof("Model published in full sync %s", modelName)
+		nodes.PublishKeyToRestLayer(modelName, "fullsync", sharedQueue)
+
+	}
+	// Now also publish the newly generated models (if any)
+	// Publish all the models to REST layer.
+	utils.AviLog.Debugf("Newly generated models that do not exist in cache %s", utils.Stringify(allModels))
+	for modelName := range allModels {
+		nodes.PublishKeyToRestLayer(modelName, "fullsync", sharedQueue)
+	}
+}
+
+func (c *GatewayController) FullSync() {
```

Comment:
FullSync gets called half an hr approximately to sync objects from controller side to refresh Rest layer model. Do we want to avoid that functionality?